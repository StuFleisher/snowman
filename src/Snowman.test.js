import Snowman from "./Snowman";
import { render, fireEvent } from "@testing-library/react";
import img0 from "./0.png";
import img1 from "./1.png";
import img2 from "./2.png";
import img3 from "./3.png";
import img4 from "./4.png";
import img5 from "./5.png";
import img6 from "./6.png";


describe("Test the Snowman component", function () {


  test("Works without crashing", function () {
    render(<Snowman
      maxWrong="6"
      images={[img0, img1, img2, img3, img4, img5, img6]}
      words={["apple"]} />);

    });

  test("Too many wrong guesses should lose", function () {
    const { container } = render(<Snowman
      maxWrong="6"
      images={[img0, img1, img2, img3, img4, img5, img6]}
      words={["apple"]} />);

      //guess wrong 6 times
    fireEvent.click(container.querySelector("[value='b']"));
    fireEvent.click(container.querySelector("[value='c']"));
    fireEvent.click(container.querySelector("[value='d']"));
    fireEvent.click(container.querySelector("[value='f']"));
    fireEvent.click(container.querySelector("[value='g']"));
    fireEvent.click(container.querySelector("[value='h']"));

    expect(container.querySelector("[value='b']")).not.toBeInTheDocument();
    expect(container.querySelector(".Snowman-lose")).toBeInTheDocument();
    expect(container.querySelector(".Snowman-lose")).toContainHTML("apple");
  });


  test("Snapshot test: loss", function(){
    const { container } = render(<Snowman
      maxWrong="6"
      images={[img0, img1, img2, img3, img4, img5, img6]}
      words={["apple"]} />);

      //guess wrong 6 times
    fireEvent.click(container.querySelector("[value='b']"));
    fireEvent.click(container.querySelector("[value='c']"));
    fireEvent.click(container.querySelector("[value='d']"));
    fireEvent.click(container.querySelector("[value='f']"));
    fireEvent.click(container.querySelector("[value='g']"));
    fireEvent.click(container.querySelector("[value='h']"));

    expect(container).toMatchSnapshot();
  })

});