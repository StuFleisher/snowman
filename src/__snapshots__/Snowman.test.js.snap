// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Snowman component Snapshot test: loss 1`] = `
<div>
  <div
    class="Snowman"
  >
    <img
      alt="6"
      src="6.png"
    />
    <p
      class="Snowman-wrong"
    >
      Wrong Guesses: 6
    </p>
    <p
      class="Snowman-word"
    >
      _
      _
      _
      _
      _
    </p>
    <p
      class="Snowman-lose"
    >
      YOU LOSE! The correct word was
      apple
    </p>
  </div>
</div>
`;
